tests/e2e/query_modules_storage_modes/test_query_modules_storage_modes.py

```diff
@@ -0,0 +1,281 @@
+    )
+    deleter.start()
+
+    cursor.execute(
+        f"""MATCH (n)
+        CALL {api}_api.pass_node_with_id(n)
+        YIELD node, id
+        RETURN node, id;"""
+    )
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 0
+
+
+def test_deleted_node(cursor):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    deleter = Process(target=modify_graph, args=("MATCH (n:Component {id: 'A7422'}) DETACH DELETE n;",))
+    deleter.start()
+
+    cursor.execute(
+        """MATCH (n: Component {id: 'A7422'})
+        RETURN python_api.pass_node(n);"""
+    )
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 1 and result[0][0] is None
+
+
+def test_deleted_relationship(cursor):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    deleter = Process(
+        target=modify_graph,
+        args=("MATCH (:Component {id: 'A7422'})-[e:PART_OF]->(:Component {id: '7X8X0'}) DELETE e;",),
+    )
+    deleter.start()
+
+    cursor.execute(
+        """MATCH (:Component {id: 'A7422'})-[e:PART_OF]->(:Component {id: '7X8X0'})
+        RETURN python_api.pass_relationship(e);"""
+    )
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 1 and result[0][0] is None
+
+
+def test_deleted_node_in_path(cursor):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    deleter = Process(target=modify_graph, args=("MATCH (n:Component {id: 'A7422'}) DETACH DELETE n;",))
+    deleter.start()
+
+    cursor.execute(
+        """MATCH path=(n {id: 'A7422'})-[e]->(m)
+        RETURN python_api.pass_path(path);"""
+    )
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 1 and result[0][0] is None
+
+
+def test_deleted_relationship_in_path(cursor):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    deleter = Process(
+        target=modify_graph,
+        args=("MATCH (:Component {id: 'A7422'})-[e:PART_OF]->(:Component {id: '7X8X0'}) DELETE e;",),
+    )
+    deleter.start()
+
+    cursor.execute(
+        """MATCH path=(n {id: 'A7422'})-[e]->(m)
+        RETURN python_api.pass_path(path);"""
+    )
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 1 and result[0][0] is None
+
+
+def test_deleted_value_in_list(cursor):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    deleter = Process(
+        target=modify_graph,
+        args=("MATCH (:Component {id: 'A7422'})-[e:PART_OF]->(:Component {id: '7X8X0'}) DELETE e;",),
+    )
+    deleter.start()
+
+    cursor.execute(
+        """MATCH (n)-[e]->()
+        WITH collect(n) + collect(e) as list
+        RETURN python_api.pass_list(list);"""
+    )
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 1 and result[0][0] is None
```

Comment:
Is deleted edge always goes first - `result[0][0]`? Based on code, I would expect to have nodes at the beginning of list

Reply:
Should a function return a deleted value, or an object containing a deleted value, it would return `None` instead. A list comprised of [node, deleted_edge] falls under the latter case.